components/PsychrometricCharts.py

```diff
@@ -0,0 +1,177 @@
+import math
+import numpy as np
+import plotly.graph_objs as go
+from dash import Dash, dcc, html
+from dash.dependencies import Input, Output
+
+# 计算函数
+def calculate_saturation_vapor_pressure(temp_c):
+    return 610.94 * np.exp(17.625 * temp_c / (temp_c + 243.04))
+
+def calculate_dew_point(temp_c, relative_humidity):
+    saturation_vapor_pressure = calculate_saturation_vapor_pressure(temp_c)
+    actual_vapor_pressure = relative_humidity / 100.0 * saturation_vapor_pressure
+    dew_point = 243.04 * np.log(actual_vapor_pressure / 610.94) / (17.625 - np.log(actual_vapor_pressure / 610.94))
+    return dew_point
+
+def calculate_wet_bulb_temp(dry_bulb_temp, relative_humidity):
+    epsilon = 0.0001
+    wet_bulb_temp = dry_bulb_temp
+    while True:
+        prev_wet_bulb_temp = wet_bulb_temp
+        wet_bulb_temp = dry_bulb_temp * math.atan(0.151977 * math.sqrt(relative_humidity + 8.313659)) \
+                        + math.atan(dry_bulb_temp + relative_humidity) \
+                        - math.atan(relative_humidity - 1.676331) \
+                        + 0.00391838 * (relative_humidity ** 1.5) * math.atan(0.023101 * relative_humidity) \
+                        - 4.686035
+        if abs(prev_wet_bulb_temp - wet_bulb_temp) < epsilon:
+            break
+    return wet_bulb_temp
+
+def calculate_absolute_humidity(dry_bulb_temp, relative_humidity):
```

Comment:
this is good but you should use the function from pythermalcomfort so we do not have code repetitions

Reply:
![image](https://github.com/user-attachments/assets/6da74c77-18c9-4353-a51d-97c7a6020d1f)
the calculation is now different from before